common/components/GenerateTransaction/GenerateTransaction.tsx

```diff
@@ -32,13 +33,14 @@ class GenerateTransactionClass extends Component<StateProps> {
       networkRequestPending,
       validAmount,
       validGasPrice,
-      validGasLimit
+      validGasLimit,
+      checkValidAmount
     } = this.props;
 
     const isButtonDisabled =
       !isFullTransaction ||
       networkRequestPending ||
-      !validAmount ||
+      (checkValidAmount && !validAmount) ||
```

Comment:
`checkValidAmount` should be a parameter to the `isValidAmount` selector instead, allowing it to be used in more scenarios and easier to test for